### DISCUSSION TITLE
Spend based KRE

### DISCUSSION THREAD

**Discussion by skyf0xx** [kinfoundation/kin-rfcs#143] [Spend Based KRE]
This discussion was started based on a recent proposal

https://github.com/kinecosystem/rewards-engine/blob/master/improvement-proposals/Spend-Based-KRE.md

**Comment by skyf0xx** [comment]
Posting some discussion here on a different thread from @Freedom-kin 

> I think the more factors you consider for a KRE score, the more gaming opportunities you give.
> 
> It could be real simple.
> Take the net SPENDS (spends minus earns) and reward developers with a percentage of that.
> This with a cap of lets say 50,000,000 Kin.
> 
> e.g. net SPENDS is 1000
> Reward is 1000 x 30% = 300 Kin
> 
> That way developers have 2 sources of income...their own SPENDS and the 30% KRE reward. In this example the developer would have an income of 1000 plus 300 = 1300 Kin.
> 
> The fact that you deduct the EARNS from the SPENDS is an incentive for the developers to generate more SPENDS then EARNS.
> SPENDS are his turnover and EARNS are costs to get it.
> 
> More SPENDS than EARNS means that Kin starts FLOWING through the economy.
> 
> Developers will not solely depend on KRE. They have their own turnover in the form of SPENDS.
> 
> No other factors means no other gaming possibilities.
> 
> I think this system would be a great incentive for larger apps to join the system.
> Where do you get a 30% bonus on top of your regular net turnover?
>

**Reply by skyf0xx** [reply]
@Freedom-kin I moved the discussion here and also linked to the proposal, since it's quite detailed and also to keep the discussion centered on this piece - I will post what my thinking was below (so others can also chime in and you can also address any questions

**Reply by skyf0xx** [reply]
This method has merit, but my issue with it is that the net flow is always towards the markets.

A developer gets 1,000 Kin from users (who got it from the market) and 300 Kin from the KRE. To cover costs, 1,300 is then sold in exchange for USDC. This remains constant even as the number of apps and users grow.

This is similar to KRE 1 & 2 where developers were rewarded for transactions.

Proposals that maximize user balances and P2P (no Kin back to the developer) force Kin into apps and keep it there. With a bigger economy or fewer rewards, the developer becomes a reseller of Kin to users. To illustrate:

If the KRE only gave out 1,000 Kin per day, any Kin going in to PeerBet would have to come from exchanges and would be more than the developer is getting from the KRE. Meanwhile, users would be using that Kin for P2P bets. Kin would also be more valuable so that trickle coming in from the KRE would be enough.

The same effect would happen with many more apps and users demanding more than the current 250MM/ day inflation rate.

**Reply by skyf0xx** [reply]
This was freedom's answer to the above comment:

> 
> ### Imagine living on an island and all you can buy for USD is apples. 
> Well, how many apples can you eat? How many dollars do you think I'll buy? Enough to buy apples, but I don't need thousands of dollars in my account for that. I always buy just enough dollars to buy apples.
> 
> Compare that to the neighboring island where you can buy a thousand different products for your dollars. How many dollars do you think I'm buying now?
> 
> Now it is interesting to have a stock of dollars in my account.
> A whole economy has developed there and that makes it interesting for me to buy a lot of dollars.
> 
> What is the difference?
> The SPEND possibilities.
> 
> **On Island 1**, the seller of apples himself will also sell the dollars he received as soon as possible, because he already has enough apples and there is nothing else he can do with dollars ( **that's the problem you described** ).
> 
> **On Island 2**, the seller can use his dollars for other products and his dollar remains in circulation without having to sell it.
> 
> ### Your reasoning about the circle, exchange-user-app-exchange, is therefore only a problem on island 1.
> 
> Island 2 solved that problem by creating 1000 different SPEND possibilities. But that takes time and a policy aimed at promoting SPENDS.
> 
> That's what I'm trying to achieve with a KRE 4.0 focused on SPENDS.
> 
> 
> But what if you try to solve the problem on island 1 by handing out free dollars? You just raise the interest on the local bank and people start saving because of the high interest (**staking?**). People don't spend their dollar on apples anymore. What little economy there is will come to a complete standstill.
> 
> **On top of that, island 1 now also suffers from enormous inflation**.
> The free dollars have created no value.
> 
> **Everyone feels rich,** but they can still only buy apples that now have risen considerably in price....and they don't need more apples than they did before.
> 
> That's what will happen as a result of a KRE based on AUB. 
> 
> It's rewarding on the basis of savings (AUB).
> As soon as it is actually spent, the reward for the developer disappears. So the developer would be crazy to create a lot of SPEND options for the user and the effect is that the economy remains very small and users  have no reason to hold their Kin or buy more.
> 
> ### Everything you reward gets bigger....we need SPENDS.
> .

**Comment by ckiz** [comment]
No need for comments,  just implement this tomorrow please!

**Reply by Freedom-kin** [reply]
Thank you for the support.

**Comment by amiraliaaa** [comment]
One major problem with relying only on Net Spends is that it makes the gaming process easy. One app with only 1 user can get a high KRE score. I think at the end of the day the KRE formula needs to somehow incorporate AU or Active spenders. Otherwise any small app can easily game the system. Also not considering the user base of the app as a factor in the KRE formula, demotivates bigger apps to join the ecosystem. 
I also liked the monopoly clause more than the daily cap. The daily cap set at 50m wont make sense for some apps as they are giving out more than that in daily rewards to the users so by having this cap some developers might even lose money rather than making any profit from the KRE.

**Reply by Freedom-kin** [reply]
Hi Amiralia,

Thanks for responding. I admit it's a completely different way of thinking and I can imagine it takes some getting used to. I will try to answer point by point. 

If 1 app sells a sports car to 1 user, that is an enormous turnover. However, turnover is turnover. The baker who sells 100 loaves of bread for a few $$ each is not better just because he has 100 buyers. 

Other apps don't get less because 1 app has a high turnover.
Moreover, if someone manages to artificially increase his turnover, he will have a problem with the tax authorities. He's going to pay taxes on sales he didn't really have. Not smart.


There is no need to count AU or AS. The more factors you take into account, the more susceptible it becomes to fraud. 

I don't understand why it would demotivate bigger apps. They are large because they apparently have a good revenue model. They must have SPENDS to stay big. With this proposal they just get a 30% bonus on top of that. How can that be demotivating?


A monopoly clause is no longer necessary because there is no competition between apps anymore. A large app does not have any negative effect on a smaller one. They each get 30% of their net sales, however much or however little that is. 

You say that some apps spend more than they will receive and that's why the 50MM cap makes no sense. There are only two apps that receive more than 50MM Kin a day average. Isn't it true that you now also adjust your payouts to what you receive from KRE? That's why the APY keeps getting lowered.
The whole idea is to get inflation under control. The amount of free Kin should be reduced.

If you subtract all APY from the KRE received, how much are you left with? I don't think it's a lot. On balance, it's a matter of transferring KRE money to users...and that was never the intention of the KRE.

However, apps had no other choice. If they didn't they couldn't compete for KRE. That is the disadvantage of the current KRE.

The last point is that they are not making any profit from the KRE. That is also a change in this proposal. KRE will be a kind of subsidy, a supplement to your income, an incentive. It is not the intention that you have 100% income from the KRE. That is not sustainable. Who should pay that?

Keep in mind that all Kin issued by the KRE is ultimately paid by all Kin holders in the form of inflation. 

As the proposal says, we will have to go back to the basics of economic thinking for now.

This proposal gives you 2 sources of income. Your own SPENDS and the KRE reward as a bonus.

**Comment by amiraliaaa** [comment]
Hi Freedom-Kin,

I have a few comments regarding your reply:

- You said "Moreover, if someone manages to artificially increase his turnover, he will have a problem with the tax authorities. He's going to pay taxes on sales he didn't really have. Not smart". 
This logic can be applied to all the KRE formulas even to the old KRE formula. However, I think one of the most important things in the KRE formula is to make it hard for developers to game the system. This new formula makes it SUPER easy for developers to game the system with only one wallet. Im sure developers who manage to game the system can also find a way to avoid taxes so making the KRE hard to be gamed should be one of the number one priorities.

- "The more factors you take into account, the more susceptible it becomes to fraud"
I actually think the opposite. The more factors you add it makes it harder for the developers to game the system. 

- "This proposal gives you 2 sources of income. Your own SPENDS and the KRE reward as a bonus."
I think this logic again can be applied to all the KRE formulas. Even with the current one the developers' sources of income are the net SPENDS and the KRE reward

- "I don't understand why it would demotivate bigger apps. "
Bigger apps get demotivated to join the ecosystem when they find out its super easy for smaller developers and smaller apps to game the system and get more reward than them. 

I personally prefer the current KRE formula or even the 3.1 version, as it also incorporates AU which I think should be the most important factor in the KRE formula.

**Reply by Freedom-kin** [reply]
I don't think it can be applied to all KRE formulas.
If you artificially increase your KRE 3.0 score you get x amount Kin. That is real income that you actually get paid.

KRE 4.0 rewards with 30%.
If you have 1000 SPENDS you get 300 Kin.   
Total income is 1300 Kin.   
Tax rates differ per country, but lets say VAT = 20% and income tax is 35%

1300 / 120 *20         = 216  (VAT)   
(1300 – 216) * 35%    = 379  (TAX)   
                       ----------------   
Total tax 216 + 379 = 595   
Net income 1300 – 595 =  705 Kin   


Now you tripple your net turnover by gaming your spends.
3900 / 120 * 20	  =    650  (VAT)
(3900 – 650) * 35%  = 1137   (TAX)
				 -----------------
Total tax 650 + 1137 = 1787
However, you didn't really got 3900 turnover, that was fake.
You had just 1300.
Net income is now 
1300 – 1787 =  - 487  you now actually have a loss

There are no other factors to game, so I think  that answers your second point.

Point 3.
Apps hardly have any real spends. 
The incentive of KRE 3.0 is to have as less as possible spends.
Every SPEND is at the expense of the AUB ( = less KRE score).
So theoretically you might be right, but in practice your main income is from the KRE.

Point 4
I think my answer to point 1 already answered this.
It doesn't bring you much profit to game the system.
And even if it could be done, other apps will not get  less KRE.
They get their 30% as promised.
There is no competition between apps anymore.

I Think this is a great advantage.
As an app you can do very well now, but if a very large app joins the system your KRE payout becomes a lot lower.
In fact, it is in the interest of developers to keep the amount of apps as small as possible.  The total amount of available KRE stays constant, so the less apps the larger your piece of the pie.

That problem is solved in KRE 4.0

**Comment by KinLover2** [comment]
I think discussing these points without clarifying some claims in the discussion is not going to be productive and in general completely redesigning the KRE normally results in more loss of current partner participation rather than helping the ecosystem as proven historically.
 
“Where do you get a 30% bonus on top of your regular net turnover?”

It should be clear why a big company is not looking at this the way it’s looked at here. A big company with a sustainable (dollar based) business model has no incentive to tie its earnings to a super volatile asset whose value has changed in multiples (up or down) over the past year alone. It is nearly impossible to risk earnings and tie it to such volatility. A sustainable business model will never take such a chance and fully commit to Kin as their main currency, even if they are earning a 30% return ON THEIR KIN, this could translate to loss of revenue because of Kin devaluing. The best such a company does is to leverage KRE for some bonus revenue as opposed part of their main strategy where they deliver no value to the ecosystem.  

Where Kin has been successful is building apps from ground up, like Peerbet, while not a perfect use case of Kin, it’s a great example of how Kin can power up a company and create revenue for it, making it sustainable. 

Companies must pay for user acquisition and expecting big companies to align their entire strategy for Kin Ecosystem (making it part of their daily announcements, giving it a prominent presence in their platform, etc.) for free is not reasonable. We are asking the big guys (who’ve potentially spent millions in getting users) to not only risk their revenue, but to share their acquired users for free. 

We need novel use cases, done by small, medium, or large companies to make owning and transacting in Kin valuable where experiences can truly be compared to each other in terms of value delivered to users, ecosystem, and partner. **This is achieved by giving Active Spenders the main role** as they basically represent 1) a retained user by the platform and 2) retained transaction within the Kin ecosystem and 3) the user benefits from exposing more people to Kin as their wallet gets larger. 
This is all done with minimal risk to the user, ecosystem or platform and the Kin ecosystem does not center around companies but experiences.

**Reply by Freedom-kin** [reply]
Hi Kinlover2, thanks for responding.
I will try to answer point by point.

1. _completely redesigning the KRE normally results in more loss of current partner participation_

	We have no choice but to redesign the KRE because it's not working. Giving away free Kin without creating value for the economy equals inflation. If we go on this way Kin will soon be worth nothing and how is that benefiting developers?   
	Larger apps are not interested in keeping money IN user wallets. They want money coming OUT of wallets to be spend in their app. But that is not rewarded by the KRE.


2. _big company with a sustainable (dollar based) business model has no incentive to tie its earnings to a super volatile asset whose value has changed in multiples_ 

	Agree on that. It's not realistic to expect that they take their WHOLE business and put it solely in Kins economy. So lets not expect that. Maybe some will consider extent their business.  


3. _A sustainable business model will never take such a chance and fully commit to Kin as their main currency, even if they are earning a 30% return ON THEIR KIN, this could translate to loss of revenue because of Kin devaluing._

	With this attitude we might as well stop looking for businesses to join the Kin Economy. Ask yourself why Kin is devaluing. The KRE is discouraging Economic Activity. We hardly have an economy and we just give Kin away for free. We first have to set things strait with the KRE and create an environment  that businesses would like to join.  

4. _The best such a company does is to leverage KRE for some bonus revenue as opposed part of their main strategy where they deliver no value to the ecosystem._

	Why would we give them a 30% bonus if they deliver no value to the ecosystem?  We don't need apps that only want the free Kin.
We need apps that generate SPEND opportunities.

5. _Where Kin has been successful is building apps from ground up, like Peerbet, while not a perfect use case of Kin, it’s a great example of how Kin can power up a company and create revenue for it, making it sustainable._

	The way it works now is that money is going one way.
From the Kin holders via the KRE to the developers.
Please tell me what's in it for the Kin holders? It sounds great that we build apps from ground up, but who is paying the bill?
Again it's the Kin holders, investors, who pay for it.
How is this sustainable?  And what happens if you take away the KRE?  Will they still be sustainable?



6. _Companies must pay for user acquisition and expecting big companies to align their entire strategy for Kin Ecosystem (making it part of their daily announcements, giving it a prominent presence in their platform, etc.) for free is not reasonable_

	Nobody is asking that from them. This is putting things to the extreme and I'm not sure why you do that.

7. _We are asking the big guys (who’ve potentially spent millions in getting users) to not only risk their revenue, but to share their acquired users for free._

	Again, nobody is asking that from them.

8. _We need novel use cases, done by small, medium, or large companies to make owning and transacting in Kin valuable where experiences can truly be compared to each other in terms of value delivered to users, ecosystem, and partner._

	Agreed, but we don't have them, and if we don't change we will not get them either. Only if we generate a flow of Kin through the economy Kin will have any value. Rewarding Kin IN wallets does the opposite.


9. _This is achieved by giving Active Spenders the main role as they basically represent 1) a retained user by the platform and 2) retained transaction within the Kin ecosystem and 3) the user benefits from exposing more people to Kin as their wallet gets larger._

	All of which are also achieved by rewarding net SPENDS.
SPENDS are only done by 1) Spenders. If you have compelling spend opportunities they will be 2) retained. 3) The user benefits from keeping inflation under control and higher Kin prices.   However, I don't see the importance of factors 1 + 2 to base the KRE on.   
If you are on vacation and buy something in a store you will probably never come back there. Does your spent money now mean less  because you are not an AU or is not retained?

A spend is a spend, regardless of how many times a user comes back. Make your spend opportunities compelling enough and they will come back.

At the end of the day the baker counts his money.
He doesn't keep a record to whom he sold his loaves of bread.


What you are basically advocating is keeping the KRE as it is.


10. _This is all done with minimal risk to the user, ecosystem or platform and the Kin ecosystem does not center around companies but experiences._

	I have to disagree on this one. As you said yourself, Kin devaluing is a reality and this is an eminent threat to Users and the whole ecosystem.  I think it's a misconception to think it's not about companies (apps). Without them there is no economy.

Giving away Kin might be a great experience, but it's not building an economy. 

Are you in any way affiliated to Peerbet?

**Comment by ted-livingston** [comment]
Hey Freedom. I appreciate you taking the time to put this proposal together. It gets me excited to see you and others leaning in on this, as I agree the KRE is the most important part of Kin. I hope we can all work together to explore these ideas and find the best solutions.

I have a few thoughts on your proposal, but I wonder if it might be helpful to take them one at a time. My goal isn’t to be pedantic, but to keep the discussion focused so that we aren’t trying to tackle multiple topics at once.

Perhaps my first question is around how you define “earn” and “spend”.

For example, is a buy through an in app buy module an earn? Is a deposit from an exchange?

For a spend, is it anything that lowers a user’s balance? If not does a user to user transaction within an app count? Does a withdraw to an exchange? What about a transfer from one app to another?

I think these definitions are important to both reduce gaming and create the right incentives.

For example, one way how I see gaming could potentially occur:

1. Developer buys 1000 Kin on an exchange
2. Developer deposits 1000 Kin from the exchange into “user_account1”, which they control
3. user_account1 tips 1000 Kin to user_account2, another account controlled by the developer
4. user_account2 withdraws 1000 Kin to an exchange, where the developer sells it

In this scenario it seems like user_account1 might look like it had a net spend of 1,000 Kin, resulting in a 300 Kin bonus for the developer, even though they are just moving money around programmatically. Perhaps even simpler, instead of tipping user_account2, the developer could perhaps just have user_account1 buy a digital good from the app. There likely wouldn't be any sales tax, as just because an app tells the KRE it made a sale, it seems they could tell their accountant they were just moving money around, which I don't think would be breaking any accounting rules.

What are your thoughts on this and how your proposal might protect against it?

**Reply by Freedom-kin** [reply]
Thank you Ted, for responding.   
I agree, the KRE can be the backbone of Kin and that's why I think getting it right is so important.

Taking one problem at the time is a good idea.
Maybe we can start a new discussion for each problem.

May I suggest you copy your comment to a new discussion?
It might be easier for others to jump in the conversation.
I'll be happy to answer you.

**Reply by Freedom-kin** [reply]
Ted, I've taken the liberty of copying your comment into a new discussion.

https://github.com/kinecosystem/rewards-engine/discussions/152
